src/js/models/_markerable.js

```diff
@@ -11,7 +12,11 @@ export default class Markerable extends Section {
     this.isMarkerable = true;
     this.tagName = tagName;
     this.markers = new LinkedList({
-      adoptItem: m => m.section = m.parent = this,
+      adoptItem: m => {
+        assert(`Cannot insert non-marker into markerable (was: ${m.type})`,
+               m.isMarker);
+        m.section = m.parent = this;
+      },
```

Comment:
This is one of the reasons I dislike the implicit return on `=>` without `{`. Was the return value important here and this refactor is in error? It is difficult to know when reading.

But I know from experience it is not in error. Just a note to share.